ecc_enclave/enclave/enclave.cpp

```diff
@@ -115,10 +145,11 @@ int ecall_cc_invoke(const uint8_t* signed_proposal_proto_bytes,
         crm = {};
 
         {  // encrypt response
-            b = g_cc_data->encrypt_message(return_encryption_key,
-                ByteArray(b64_response.c_str(), b64_response.c_str() + b64_response.length()),
-                encrypted_response);
-            COND2LOGERR(!b, "cannot encrypt message");
+            ByteArray response =
+                ByteArray(b64_response.c_str(), b64_response.c_str() + b64_response.length());
+            CATCH(b, encrypted_response =
+                         pdo::crypto::skenc::EncryptMessage(response_encryption_key, response));
```

Comment:
The decryption and the encryption operations were in `cc_data` 
* to keep them in a single place, together with the crypto keys
* and to confine the pdo crypto usage to `cc_data`

This breaks the abstraction, and the pdo libs are accessible only because they had to be exposed in `cc_data.h`, otherwise they would not be available.
This part should be reverted.

Reply:
I am not yet convinced about reverting this part.

I like the abstraction that all the enclave keys are not exposed outside `cc_data`. However, this abstraction is already broken in `shim.cpp` where the `state_encryption_key` is used with the pdo crypto functions. Also `enclave.cpp` used `pdo::crypto::constants::SYM_KEY_LEN` before.

The methods to decrypt and encrypt the request and the response messages using the keys from the `keyTransportMessage` are, I believe, not related to the enclave state (cc_data), that is, they are not consuming any state of `cc_data`, and thus, for me `cc_data` is the wrong place to provide such an abstraction for the pdo crypto.

In contrast, to limit the scope of the chaincode_decryption_key used to decrypt KeyTransportMessage it is OK for me to have `decrypt_key_transport_message` as part of `cc_data`. Simply because this function depends on the state.

A clean solution, I believe, would be to have a `crypto.cpp/h` in `common/crypto` which offers all primitives/constants needed in FPC, so we can remove all references to `pdo::crypto` elsewhere. I also think this is beyond the scope of this PR.

Reply:
Oh we already have `pdo-crypto-c-wrapper.h/cpp`. Can't we just use this?

Reply:
> Oh we already have pdo-crypto-c-wrapper.h/cpp. Can't we just use this?

I don't see why we can't, but in my opinion we should not.
As I said above, this wrapper additionally makes a translation between object-oriented and procedural abstractions.
This comes with some overhead at the edge due to data serialization/deserialization and object build/destruction.

Reply:
ok, I moved the static (generic) methods for message encryption/decryption to `enclave/crypto.cpp` and extended cc_data with methods to encrypt/decrypt state values without returning the state_encryption_key to `shim.cpp`. That is, `cc_data` not provides all crypto functions that are related to it's state. `cc_data` still uses pdo crypto directly. Stateless, crypto functions such as message encryption/decryption/hashing are not available in `crypto.cpp` and can be used wherever needed.

Eventually, the dependency to pdo crypto can be removed also from `cc_data` by wrapping pdo methods inside `crypto.cpp`. I think we have now a cleaner solution and did a bit more of refactoring as needed to solve the actual problem - but this is also not a bad thing. Let's aim to finish this PR and conclude this discussion now as it is growing.